tools/checklocks/state.go

```diff
@@ -332,10 +339,18 @@ func (l *lockState) valueAsString(v ssa.Value) string {
 
 // String returns the full lock state.
 func (l *lockState) String() string {
-	if l.count() == 0 {
+	if l.numLocksHeld() == 0 {
 		return "no locks held"
 	}
-	return strings.Join(l.lockedMutexes, ",")
+
+	var locksHeld []string
+	for k, lockType := range l.lockedMutexes {
+		if lockType == locked {
+			locksHeld = append(locksHeld, k)
+		}
+	}
+
```

Comment:
Who uses String()? Is it just human readable & tests?

Reply:
It's used in error reporting for the current locks held.

Reply:
Not sure whether we should sort or not. I am guessing yes to removed some noise on comparing results.